sdk/src/main/java/com/hcaptcha/sdk/HCaptchaHeadlessWebView.java

```diff
@@ -46,24 +46,24 @@ final class HCaptchaHeadlessWebView implements IHCaptchaVerifier {
     public void startVerification(@NonNull FragmentActivity activity) {
         if (webViewLoaded) {
             // Safe to execute
-            resetAndExecute();
+            webViewHelper.resetAndExecute();
         } else {
             shouldExecuteOnLoad = true;
         }
     }
 
-    private void resetAndExecute() {
-        webViewHelper.getWebView().loadUrl("javascript:resetAndExecute();");
-    }
-
     @Override
-    public void onFailure(final HCaptchaException exception) {
-        final boolean silentRetry = webViewHelper.getConfig().getResetOnTimeout()
-                && exception.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT;
-        if (silentRetry) {
-            resetAndExecute();
+    public void onError(@NonNull final HCaptchaError error) {
```

Comment:
Why the rename from `onFailure` to `onError`?

Reply:
In this implementation `HCaptchaException` keep a reference on `HCaptchaRetryer` (it's passed via constructor). So I decided to move `HCaptchaException` construction to `HCaptchaDialogFragment/HCaptchaHeadlessWebView` level.

But that's not the crucial point. I think make sense to try move `HCaptchaException` back it will reduce code duplication